### ISSUE TITLE
[NOD - Accessibility]: Request an extension markup and content improvements

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/va.gov-team#76909]
## Value Statement
**_As a_** person who uses a screen reader
**_I want to_** be able to here critical information when navigating form fields alone 
**_So that_** I can provide the needed information and make the best decisions the questions.

---
## Background

While talking to @briandeconinck about the upcoming staging review for NOD, and the [remaining accessibility item from the midpoint review](https://github.com/department-of-veterans-affairs/va.gov-team/issues/76482), I found an oddity in the code for one of the screens.

The current code for the "[Request an extension](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/extension-request)" page:

```html
<fieldset class="vads-u-margin-y--2 rjsf-object-field">
  <div>
    <fieldset class="vads-u-margin-y--2 rjsf-object-field">
      <legend class="schemaform-block-title schemaform-block-subtitle" id="root_view:requestExtensionInfo__title"></legend>
      <h3 class="vads-u-margin-top--0">Request an extension</h3>
      <p class="vads-u-margin-y--0">If you request an extension (extra time) to file VA Form 10182 for good cause, you'll need to tell us why you have good cause.</p>
    </fieldset>
  </div>
  <div>
    <va-radio class="rjsf-web-component-field hydrated" enableanalytics="true" label="Are you requesting an extension?" name="root_requestingExtension" required="false" uswds="true" value="true" aria-invalid="false"> ... </va-radio>
  </div>
</fieldset>
```

The legend here is empty. It should have the h3 (`<h3 class="vads-u-margin-top--0">Request an extension</h3>`) in it.

And the paragraph text (`<p class="vads-u-margin-y--0">If you request an extension (extra time) to file VA Form 10182 for good cause, you'll need to tell us why you have good cause.</p>`) could be moved up as hint text. The "you'll need to tell us" might be helpful for screen readers to here when browsing. When screen readers browse forms, they may not catch all the content outside of the fields/labels.

We should talk through the best solution. It might be moving the text up or reworking the content somehow to be clearer and shorter.

## Acceptance Criteria
- [ ] Users with screen readers can clearly hear the "good cause" content.
- [ ] Fix verified in a11y review of staging

## Tasks
- [ ] Remove nested fieldset/legend
- [ ] Move header & hint text into legend of fieldset wrapping radios
- [ ] a11y testing
- [ ] Check unit tests

## Definition of Ready 
- [ ] Clear value description
- [ ] Testable acceptance criteria
- [ ] Accessibility added to acceptance criteria
- [ ] Approved designs attached
- [ ] Sample data provided where appropriate
- [ ] Estimated to fit within the sprint
- [ ] Dependencies and blockers linked

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Reviewed and approved by product and/or design
- [ ] Reviewed and signed off by a11y

**Comment by Mottie** [comment]
@davidakennedy This work has been merged in and ready for testing

**Comment by davidakennedy** [comment]
I tested this in staging:

- Everything works as expected on JAWS and NVDA. The code looks good.
- This does lead to multiple announments of the content in VoiceOver on Mac because of the way it announces "grouped" content in legends. That's expected, and is a worthy tradeoff for folks who use other screen readers.

**Comment by saderagsdale** [comment]
@davidakennedy Is the "good cause" content easy to hear? If so, I think we can close this.

**Comment by davidakennedy** [comment]
@saderagsdale Yes. You hear it in order (heading, then hint/paragraph text) when JAWS or NVDA first enters a fieldset, either going forward or backward.
